converter/service/converter.py

```diff
@@ -109,12 +115,14 @@ def handle(decoded_url, user_agent, accept):
     try:
         res = urlopen(req, timeout=MAX_TIMEOUT)
         charset = res.info().get_param(param_charset) or default_charset
-        raw_data = res.read(MAX_BYTES+1)
-        if len(raw_data) > MAX_BYTES:
-            logging.info('%s response exceeded the size limit %d',
-                         decoded_url,
-                         len(raw_data))
+
+        if header_content_length not in res.headers:
+            raise ValueError(ErrorResponse.ENDPOINT_HEADER_MISS.to_msg())
+
+        if int(res.headers.get(header_content_length)) > MAX_BYTES:
             raise ValueError(ErrorResponse.MAX_SIZE.to_msg())
+
+        raw_data = res.read(MAX_BYTES)
```

Comment:
We already checked the `content-length`. You can just do `raw_data = res.read()`.